lib/src/component_declaration/component_base.dart

```diff
@@ -175,7 +166,10 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component {
   @override
   TProps get props {
     var unwrappedProps = this.unwrappedProps;
-    TProps typedProps = _typedPropsCache[unwrappedProps];
+    /// Have to cast as [TProps] until we can parameterize [_typedPropsCache].
```

Comment:
Hm, I wonder if we could reintroduce the type parameter on this Expando to avoid this cast... it seems the linked issue might only apply to mixins, so the omitted parameterization may just be carried over from when this code was part of a mixin. 

These getters are used a lot, so it would be nice to avoid the perf hit of this cast if possible.